app/javascript/components/search/controls/CombinedFacetControl.js

```diff
@@ -0,0 +1,123 @@
+  const [showFilters, setShowFilters] = useState(false)
+  const facetContext = useContext(SearchFacetContext)
+  const selectionContext = useContext(SearchSelectionContext)
+  const searchContext = useContext(StudySearchContext)
+
+
+  const facetContents = _compact(facetIds.map(facetId => {
+    const facet = _find(facetContext.facets, { id: facetId })
+    if (!facet) {
+      return null
+    }
+    const selection = getSelectionForFacet(facet, selectionContext)
+    return { facet, selection }
+  }))
+
+  const { node, handleButtonClick } = useCloseableModal(showFilters, setShowFilters)
+
+  const canApply = facetContents.map(facetContent => {
+    return isFacetApplicable(facetContent.facet, selectionContext, searchContext)
+  }).some(canApply => {return canApply})
+
+  const canClear = facetContents.map(facetContent => {
+    return facetContent.selection.length > 0
+  }).some(canClear => {return canClear})
+
+  // if any of the contained facets have selections, indicate the number in the control name
+  const appliedSelection = _compact(_flatten(facetIds.map(facetId => searchContext.params.facets[facetId])))
+  let appliedControlName = controlName
+  let hasSeachedFilters = false
+  if (appliedSelection.length) {
+    appliedControlName = `${controlName} (${appliedSelection.length})`
+    hasSeachedFilters = true
+  }
+
+  /** clear the selection, don't perform the search */
+  function clearFilters() {
+    const updatedSelection = { facets: _clone(searchContext.params.facets) }
+    facetContents.forEach(facetContent => {
+      updatedSelection.facets[facetContent.facet.id] = []
+    })
+    selectionContext.updateSelection(updatedSelection)
+  }
+
+  /** do the search and close the modal */
+  function handleApplyClick() {
+    if (!canApply) return
+    selectionContext.performSearch()
+    if (setShowFilters) {
+      setShowFilters(false)
+    }
+  }
+
+  if (!facetContents.length) {
+    // either this instance doesn't have the facets populated, or the facets havent finished loading yet
+    return <span></span>
+  }
+
+  return (
+    <span ref={node} className={`facet ${showFilters ? 'active' : ''} ${hasSeachedFilters ? 'selected' : ''}`}>
+      <a onClick={handleButtonClick}>
+        {appliedControlName}
+      </a>
+      {
+        showFilters && <div className="filters-box-searchable combined-facet">
+          <div className="multi-facet-container">
+            { facetContents.map((facetContent, index) => {
+              return facetContent.facet &&
+                <div className="single-facet" key={facetContent.facet.id}>
+                  { /* only show the facet names if there are more than one */ }
+                  { facetContents.length > 1 && <h4>{facetContent.facet.name}</h4> }
+                  <FiltersBoxSearchable
+                    show={showFilters}
+                    facet={facetContent.facet}
+                    setShow={ () => {} }
```

Comment:
This looks extraneous, at a glance.

Reply:
the idea is that filtersBoxSearchable expects a function for this argument, and so I'm passing it a no-op.  The alternative is to add checks before any calls in filtersBoxSearchable to setShow that setShow exists. I could see either one being correct

Reply:
Agreed, either works.  Thanks for helping me understand here.